x/asset/module/module.go

```diff
@@ -0,0 +1,101 @@
+package module
+
+import (
+	"context"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/spf13/cobra"
+
+	"cosmossdk.io/core/appmodule"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	cdctypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+
+	"github.com/provenance-io/provenance/x/asset/client/cli"
+	"github.com/provenance-io/provenance/x/asset/keeper"
+	"github.com/provenance-io/provenance/x/asset/simulation"
+	"github.com/provenance-io/provenance/x/asset/types"
+	"github.com/provenance-io/provenance/x/exchange"
+)
+
+var (
+	_ module.AppModuleBasic      = (*AppModule)(nil)
```

Comment:
issue: This should be asserting that the `AppModuleBasic` struct satisfies the `module.AppModuleBasic` interface.

```suggestion
	_ module.AppModuleBasic      = (*AppModuleBasic)(nil)
```

I think some of our modules have this wrong, but we should do it right in here.

Reply:
:heavy_check_mark: